coderd/database/migrations/000188_workspace_agent_port_sharing.up.sql

```diff
@@ -0,0 +1,25 @@
+CREATE TABLE workspace_agent_port_share (
+	workspace_id uuid NOT NULL REFERENCES workspaces (id) ON DELETE CASCADE,
+	agent_name text NOT NULL,
+	port integer NOT NULL,
+	share_level integer NOT NULL
```

Comment:
This needs to be an enum. We already have an enum for sharing level so you should use that instead.